app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt

```diff
@@ -268,6 +273,30 @@ private class ExternalNavController(
         }
     }
 
+    /**
+     * Navigate to Calendar Registration
+     * @param timeTableItem カレンダー登録に必要なタイムラインアイテムの情報
+     */
+    fun navigateToCalendarRegistration(timeTableItem: TimetableItem) {
+        val calendarIntent = Intent(Intent.ACTION_INSERT).apply {
+            data = CalendarContract.Events.CONTENT_URI
+            putExtras(
+                bundleOf(
+                    CalendarContract.EXTRA_EVENT_BEGIN_TIME to timeTableItem.startsAt.toEpochMilliseconds(),
+                    CalendarContract.EXTRA_EVENT_END_TIME to timeTableItem.endsAt.toEpochMilliseconds(),
+                    CalendarContract.Events.TITLE to "[${timeTableItem.room.name.currentLangTitle}] ${timeTableItem.title.currentLangTitle}",
+                    CalendarContract.Events.EVENT_LOCATION to timeTableItem.room.name.currentLangTitle,
+                ),
+            )
+        }
+
+        runCatching {
+            context.startActivity(calendarIntent)
+        }.onFailure {
+            Log.e("navigateToCalendarRegistration", "Fail startActivity", it)
```

Comment:
Thanks your review!
I think you're right. I will correct that!

Reply:
(Not 登録 but 起動 might be a more accurate representation in case of failure 🤔
But I think users can understand what happened easily whichever message is displayed. Please apply your favorite message! 🙌)

Reply:
For your reference, we are using Kermit for logger.

Reply:
Making snackbar from here may be difficult. I think you don't have to add snackbar if it is difficult 🙏

Reply:
@takahirom Thanks your comment!
I was thinking of passing the controller function in the ViewModel.
It seems to be a dirty code, so I will fix only the log 🙏

Reply:
I fixed the log with Kermit, so please review 🙏 
[`40ff9ff` (#798)](https://github.com/DroidKaigi/conference-app-2023/pull/798/commits/40ff9ff7afadf9c471612ca55e819072f4f6879e)

Reply:
Oh, I thought runCaching can return boolean result, so the value can be used in TimetableItemDetailBottomAppBar  to handle snackbar.
But it is my just idea, so no problem at all! Thank you for your consideration 🙇‍♂️

Reply:
As far as I know, currently, we are doing navigation in KaigiApp so we can't see SnackbarState 💭

Reply:
Yes, I was wondering if we could use the UserMessageStateHolder of ViewModel.